### ISSUE TITLE
Apache server not rendering django application

### ISSUE THREAD

**Issue by Divya-1019** [GrahamDumpleton/mod_wsgi#711]
Apache with SSL virtualhost and django app, Giving "This site can’t provide a secure connection"
Added following to httpd.conf[Included https-ssl(without any changes)]

    SSLSessionCache "shmcb:C:/Apache24/logs/ssl_scache(512000)"
       
    Listen 80
    Listen 443
    
    LoadFile "c:/python39/python39.dll"
    LoadModule wsgi_module "c:/python39/lib/site-packages/mod_wsgi/server/mod_wsgi.cp39-win_amd64.pyd"
    WSGIPythonHome "c:/python39"
    WSGIPythonPath "C:/tdid/DSS/"
    ServerName xx-xxx.xxxx.com
    
    <VirtualHost *:80>
    
    ServerName xx-xxx.xx.com:80
    ServerAlias www.xx-xx.xx.com
    ServerAdmin xx@xx.com
    DocumentRoot C:/tdid/DSS/
    
    WSGIScriptAlias / C:/tdid/DSS/DSS/wsgi.py
    <Directory "C:/tdid/DSS/DSS/">
        <Files wsgi.py>
            Require all granted
        </Files>
    </Directory>
    
    Alias /static "C:/tdid/DSS/static"
    <Directory "C:/tdid/DSS/static/">
        Require all granted
    </Directory>
    
    </VirtualHost>
    
    
    
    <VirtualHost *:443>
    ServerSignature Off
    DocumentRoot C:/tdid/DSS/ 
    ServerName xx-xx.xx.com:443
    
    SSLEngine On
    SSLCertificateFile "C:/Apache24/conf/xx-xx.crt"
    SSLCertificateChainFile "C:/Apache24/conf/xxx.crt"
    SSLCertificateKeyFile "C:/Apache24/conf/xx-xxx.key"
    
    WSGIScriptAlias / C:/tdid/DSS/DSS/wsgi.py
    <Directory "C:/tdid/DSS/DSS/">
        <Files wsgi.py>
            Require all granted
        </Files>
    </Directory>
    
    Alias /static "C:/tdid/DSS/static"
    <Directory "C:/tdid/DSS/static/">
        Require all granted
    </Directory>

    </VirtualHost>

Apache error log shows,

    [ssl:warn] [pid 6124:tid 132] AH01916: Init: (xxxx-xxxxx.xxxx.com:443) You configured HTTP(80) on the standard HTTPS(443) port!

Also, when I give https://domain name, I get:

    This site can’t provide a secure connection xxx-xx.xxx.com sent an invalid response.
    Try running Windows Network Diagnostics.
    ERR_SSL_PROTOCOL_ERROR

I am so stuck with this, any suggestions.

Thanks in advance.

**Comment by GrahamDumpleton** [comment]
You shouldn't include port numbers in ``ServerName`` for a start. Should be just the hostname. Try fixing that and see what then happens.

**Comment by Divya-1019** [comment]
Hi @GrahamDumpleton,

Thanks for your response.
I tried with virtualhost:80 and virtualhost:443 define and I'm getting 400 226 error on access.log.
On accessing domain name with https, I'm getting: This site can’t provide a secure connection.
Below id the httpd.conf file:

[SSLSessionCache "shmcb:C:/Apache24/logs/ssl_scache(512000)"

Listen 80
Listen 443


LoadFile "c:/python39/python39.dll"
LoadModule wsgi_module "c:/python39/lib/site-packages/mod_wsgi/server/mod_wsgi.cp39-win_amd64.pyd"
WSGIPythonHome "c:/python39"
WSGIPythonPath "C:/tdid/DSS/"
ServerName utilitiesTandD-gridinvestment.accenture.com

<VirtualHost *:80>

ServerName utilitiesTandD-gridinvestment.accenture.com
ServerAlias www.utilitiesTandD-gridinvestment.accenture.com
ServerAdmin divya.aa.gupta@accenture.com
DocumentRoot C:/tdid/DSS/

WSGIScriptAlias / C:/tdid/DSS/DSS/wsgi.py
<Directory "C:/tdid/DSS/DSS/">
    <Files wsgi.py>
        Require all granted
    </Files>
</Directory>

Alias /static "C:/tdid/DSS/static"
<Directory "C:/tdid/DSS/static/">
    Require all granted
</Directory>



</VirtualHost>



<VirtualHost *:443>
ServerSignature Off
ServerName utilitiesTandD-gridinvestment.accenture.com
ServerAlias www.utilitiesTandD-gridinvestment.accenture.com
ServerAdmin divya.aa.gupta@accenture.com
DocumentRoot C:/tdid/DSS/ 

SSLEngine On
SSLCertificateFile "C:/Apache24/conf/utilitiesTandD-gridinvestment_accenture_com.crt"
SSLCertificateChainFile "C:/Apache24/conf/DigiCertCA.crt"
SSLCertificateKeyFile "C:/Apache24/conf/utilitiesTandD-gridinvestment_accenture_com.key"

WSGIScriptAlias / C:/tdid/DSS/DSS/wsgi.py
<Directory "C:/tdid/DSS/DSS/">
    <Files wsgi.py>
        Require all granted
    </Files>
</Directory>

Alias /static "C:/tdid/DSS/static"
<Directory "C:/tdid/DSS/static/">
    Require all granted
</Directory>

</VirtualHost>](url)

I don't know what is going wrong.

Would be grateful for your advice.

Thanks.

**Comment by GrahamDumpleton** [comment]
What is in the error log? A 400 request is more likely coming from your application.

**Comment by Divya-1019** [comment]
It says:
AH01916: Init: (domain-name.com:443) You configured HTTP(80) on the standard HTTPS(443) port!

**Comment by GrahamDumpleton** [comment]
Ensure ``LogLevel`` is set to ``info`` so can at least verify that mod_wsgi is being loaded.

Also, since from the hostname it looks like you are in a corporate environment, are you sure that your browser isn't configured to use a proxy and is therefore sending the request elsewhere.

**Comment by Divya-1019** [comment]
Hi @GrahamDumpleton,

Thank you so much for the reply.
I have changed the log level and I am on a VM, proxy settings are off on IE.

Getting the following error now in log, which looks like more of a SSL certificate error(I got the certificates from "www.digicert.com", used they application to generate CSR.

Below is the log:
[Tue Aug 31 03:42:27.606127 2021] [ssl:info] [pid 15580:tid 520] AH01914: Configuring server xx-xx.xx.com:443 for SSL protocol
[Tue Aug 31 03:42:27.606127 2021] [ssl:emerg] [pid 15580:tid 520] AH02565: Certificate and private key xx-xx.xx.com:443:0 from C:/Apache24/conf/active-SSL-certificates/xx-xx.crt and C:/Apache24/conf/active-SSL-certificates/xx-xx.key do not match
AH00016: Configuration Failed

Should I use openSSL for generating CSR ?

Thanks.
